lib/spack/spack/compilers/arm.py

```diff
@@ -0,0 +1,73 @@
+# Copyright 2013-2018 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+import re
+
+from spack.compiler import Compiler, _version_cache
+from spack.util.executable import Executable
+
+
+class Arm(Compiler):
+    # Subclasses use possible names of C compiler
+    cc_names = ['armclang']
+
+    # Subclasses use possible names of C++ compiler
+    cxx_names = ['armclang++']
+
+    # Subclasses use possible names of Fortran 77 compiler
+    f77_names = ['armflang']
+
+    # Subclasses use possible names of Fortran 90 compiler
+    fc_names = ['armflang']
```

Comment:
Do people mix the arm compiler with `gfortran` or other compilers?  Or will an installation always have `flang`?  If the former, take a look at the existing `clang.py` -- we may want to add similar logic as is there so that users can mix compilers.

Reply:
The installation will always have `armflang`.